extensions/narayana-lra/runtime/src/main/java/io/quarkus/narayana/lra/runtime/NarayanaLRARecorder.java

```diff
@@ -0,0 +1,13 @@
+package io.quarkus.narayana.lra.runtime;
+
+import io.narayana.lra.client.NarayanaLRAClient;
+import io.quarkus.runtime.annotations.Recorder;
+
+@Recorder
+public class NarayanaLRARecorder {
+    public void setConfig(final LRAConfiguration config) {
+        if (System.getProperty(NarayanaLRAClient.LRA_COORDINATOR_URL_KEY) == null) {
```

Comment:
Not sure what you mean but the line of code is intentional.

Reply:
users can do `java -jar lra-quarkus.jar -Dquarkus.lra.coordinator.url=url1 -Dlra.coordinator.url=url2` which means that starting LRA with `@LRA` annotation and calling `lraClient.getAllLRAs()` in this method would not return the started LRA. `lra.coordinator.url` is `NarayanaLRAClient.LRA_COORDINATOR_URL_KEY`. So that's why I'm asking if it is intended as it seems a little strange to me. But maybe there is a use case.

Reply:
The line of code was intentional. It allows users to set the url on the java command line. But if it is going to be a blocker for this PR then I'd reluctantly remove it from the PR.

Reply:
Not a blocker but it's good to point out that this can happen. Maybe I really just don't see the use case.

Reply:
So, it can be set from a sys variable, but not an env one?  In addition, it overrides the configured value.  
Is it because the TCK expect it, or what's the reason behind it?

The configured value can be set using -Dquarkus.lra.coordinator-url and from the QUARKUS_LRA_COORDINATOR_URL env variable.

Reply:
Both of you complained about this so I guess I will have to remove it.

Reply:
I remember now why I needed it. The integration test starts the coordinator in a docker container and the port mapping needs to be exposed to the extension [1]. The javadoc for QuarkusTestResourceLifecycleManager [2] implies that it will set system properties depending on what the start method returns [3]. But this does not appear to work as expected so I set it as a system property [1] and avoid overwriting it in the extension [4].

So @cescoffier , it is required.

[1] https://github.com/mmusgrov/quarkus/blob/issue4399-lra-extension/integration-tests/narayana-lra/src/test/java/org/acme/quickstart/lra/LRAParticipantTestResourceLifecycle.java#L38
[2] https://github.com/quarkusio/quarkus/blob/main/test-framework/common/src/main/java/io/quarkus/test/common/QuarkusTestResourceLifecycleManager.java#L28
[3] https://github.com/mmusgrov/quarkus/blob/issue4399-lra-extension/integration-tests/narayana-lra/src/test/java/org/acme/quickstart/lra/LRAParticipantTestResourceLifecycle.java#L40
[4] https://github.com/mmusgrov/quarkus/blob/issue4399-lra-extension/extensions/narayana-lra/runtime/src/main/java/io/quarkus/narayana/lra/runtime/NarayanaLRARecorder.java#L9

Reply:
Doesn't `quarkus.lra.coordinator-url` work? IF your test resource returns a Map containing this property it should configure the extension with it.

Reply:
No it doesn't work.

Reply:
I mentioned this fact in my comment, quote: `The javadoc for QuarkusTestResourceLifecycleManager [2] implies that it will set system properties depending on what the start method returns [3]`. And the code already does return it.

Reply:
You need to return a `Properties` object containing the property to set. Do not set SystemPRoperty directly from TestResource.

Reply:
To repeat my last comment `I mentioned this fact in my comment, quote: The javadoc for QuarkusTestResourceLifecycleManager [2] implies that it will set system properties depending on what the start method returns [3]. And the code already does return it.` If you look at the code in the PR you will see that this is what I am doing.